extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ContextRegistrarBuildItem.java

```diff
@@ -9,12 +11,18 @@
 public final class ContextRegistrarBuildItem extends MultiBuildItem {
 
     private final ContextRegistrar contextRegistrar;
+    private final DotName annotationName;
 
-    public ContextRegistrarBuildItem(ContextRegistrar contextRegistrar) {
+    public ContextRegistrarBuildItem(ContextRegistrar contextRegistrar, DotName annotationName) {
```

Comment:
I'm not so sure it was a good idea to add the annotation name to the `ContextRegistrarBuildItem`. In theory, you can use one `ContextRegistrarBuildItem` to register multiple contexts. It might make more sense to create a separate build item or to make it `DotName[]`...

Reply:
Also it would make sense to pass `Class<? extends Annotation>` and create a `DotName`  under the hood. Maybe we should create a a builder to avoid "too many constructors hell".

Reply:
In the original issue we discussed that it would be nicer to have just one build item. IMO it's better to have one item for one task whenever possible (task being custom context registration). Otherwise you risk having a context registered but not the name of the scope.